app/models/services/service_instance.rb

```diff
@@ -110,6 +111,14 @@ def name_clashes
       end
     end
 
+    def around_save
+      yield
+    rescue Sequel::UniqueConstraintViolation => e
+      raise UniqueConstraintViolationError.new(e) if e.message.try(:include?, 'si_space_id_name_index')
```

Comment:
Maybe we just raise a `Sequel::ValidationFailed` error with the correct information:

```
if e.message.try(:include?, 'si_space_id_name_index')
  errors.add(:name, :unique)
  raise validation_failed_error
end
```

This would mean that the `validation_error!` method in the `ServiceInstanceCreateMixin` does not need to be touched at all. And this could also be a pattern for all the other models having the same problem, i.e. simply translate a `UniqueConstraintViolation` into the corresponding `ValidationFailed` error.

Reply:
sounds good